### PR TITLE
✨ feat: Breadcrumb Component Token

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Breadcrumb Component Token      |
| 🇨🇳 Chinese |    Breadcrumb Component Token       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at ef1cb5e</samp>

This pull request adds the `ComponentToken` interface to the breadcrumb component, which allows the users to customize the component style through the ConfigProvider theme prop. It also updates the documentation and the migration guide to show how to use the `ComponentToken` interface and how to map the Less variables to the ComponentToken values.

### <samp>🤖 Generated by Copilot at ef1cb5e</samp>

*  Add a component token demo for the breadcrumb component that shows how to use the ConfigProvider theme prop to customize the token values ([link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-f035605d3114d3537d76a61f5683fa22b39cc71334e46c3822e74600edbfee67R1-R7), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-3c068995759459e305b9acdf1074abfd93272faca4b2322bde93c2d2dc981cdfR1-R43), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-893bcd53601541ad21b7b34644e1606526e6a1fb419750b3494662cd190782a1R45), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-2b4604b027d70368b5422346bc06c7ed38e652ed57cd80d1ed355307d620419aR46))
*  Export the ComponentToken interface from the breadcrumb component style file and import it to the components interface file, so that the ConfigProvider theme prop can accept the ComponentToken values for the breadcrumb component ([link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-4fbc88ad11a7132e8158c1091bf3a4f6c0c2c3eab7b56e38cd6fe198fede233dL2-R17), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-1d8c7d2026da8b02ccc59970cea90849086b133e4b8f55d59ae475ac814d9e1bR7), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-1d8c7d2026da8b02ccc59970cea90849086b133e4b8f55d59ae475ac814d9e1bL63-R64))
*  Replace the redundant token properties in the `genBreadcrumbStyle` and the default export functions in the `components/breadcrumb/style/index.ts` file with the ComponentToken properties, to simplify the styling of the breadcrumb component ([link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-4fbc88ad11a7132e8158c1091bf3a4f6c0c2c3eab7b56e38cd6fe198fede233dL23-R28), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-4fbc88ad11a7132e8158c1091bf3a4f6c0c2c3eab7b56e38cd6fe198fede233dL39-R40), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-4fbc88ad11a7132e8158c1091bf3a4f6c0c2c3eab7b56e38cd6fe198fede233dL48-R49), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-4fbc88ad11a7132e8158c1091bf3a4f6c0c2c3eab7b56e38cd6fe198fede233dL56-R62), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-4fbc88ad11a7132e8158c1091bf3a4f6c0c2c3eab7b56e38cd6fe198fede233dL86-R91), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-4fbc88ad11a7132e8158c1091bf3a4f6c0c2c3eab7b56e38cd6fe198fede233dL112-R119))
*  Add the mapping table between the Less variables and the ComponentToken values for the breadcrumb component in the migration guides, to help the users to migrate from using Less variables to using ComponentToken values ([link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL22-R36), [link](https://github.com/ant-design/ant-design/pull/42342/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L22-R36))

### CODE REVIEW

File: docs/react/migrate-less-variables.en-US.md [hunk 1]
```diff
@@ -19,7 +19,20 @@ This document contains the correspondence between all the less variables related
 
 <!-- ### Badge -->
 
-<!-- ### BreadCrumb -->
+### BreadCrumb 面包屑
+
+<!-- prettier-ignore -->
+| Less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@drawer-bg` | `popoverBg` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:27]:
```suggestion
```
无关的 less 变量不用放进来

File: components/breadcrumb/style/index.ts [hunk 2]
```diff
@@ -109,14 +110,13 @@ const genBreadcrumbStyle: GenerateStyle<BreadcrumbToken, CSSObject> = (token) =>
...
-    breadcrumbLastItemColor: token.colorText,
-    breadcrumbSeparatorMargin: token.marginXS,
-    breadcrumbSeparatorColor: token.colorTextDescription,
+    baseColor: token.colorTextDescription,
+    lastItemColor: token.colorText,
+    iconFontSize: token.fontSize,
+    linkColor: token.colorTextDescription,
+    linkColorHover: token.colorText,
+    separatorColor: token.colorTextDescription,
+    separatorMargin: token.marginXS,
```
Comment (MadCcc) [inline components/breadcrumb/style/index.ts:119]: ComponentToken 要在 `genComponentStyleHook` 第三个参数初始化，不然会覆盖用户输入

File: components/breadcrumb/style/index.ts [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
-  breadcrumbLinkColor: string;
-  breadcrumbLinkColorHover: string;
-  breadcrumbLastItemColor: string;
-  breadcrumbSeparatorMargin: number;
-  breadcrumbSeparatorColor: string;
+export interface ComponentToken {
+  baseColor: string;
+  iconFontSize: number;
+  linkColor: string;
+  linkColorHover: string;
```
Comment (MadCcc) [inline components/breadcrumb/style/index.ts:10]:
```suggestion
  linkHoverColor: string;
```

File: components/breadcrumb/style/index.ts [hunk 4]
```diff
@@ -1,30 +1,31 @@
...
-  breadcrumbBaseColor: string;
-  breadcrumbFontSize: number;
-  breadcrumbIconFontSize: number;
-  breadcrumbLinkColor: string;
-  breadcrumbLinkColorHover: string;
-  breadcrumbLastItemColor: string;
-  breadcrumbSeparatorMargin: number;
-  breadcrumbSeparatorColor: string;
+export interface ComponentToken {
+  baseColor: string;
```
Comment (MadCcc) [inline components/breadcrumb/style/index.ts:7]:
```suggestion
  itemColor: string;
```

File: components/breadcrumb/style/index.ts (hunk 2 above)
Comment (MadCcc) [reply components/breadcrumb/style/index.ts:119]: image
现在 demo 上就是没有效果的

File: docs/react/migrate-less-variables.en-US.md (hunk 1 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md]: done

File: components/breadcrumb/style/index.ts (hunk 2 above)
Comment (heiyu4585) [reply components/breadcrumb/style/index.ts]: done

File: components/breadcrumb/style/index.ts (hunk 4 above)
Comment (heiyu4585) [reply components/breadcrumb/style/index.ts]: done

File: components/breadcrumb/style/index.ts (hunk 3 above)
Comment (heiyu4585) [reply components/breadcrumb/style/index.ts:10]: done

File: components/breadcrumb/demo/component-token.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            {
+              path: '/user2',
+              breadcrumbName: 'User2',
+            },
+          ],
+        },
+      ]}
+    />
+  </ConfigProvider>
+);
```
Comment (MadCcc) [inline components/breadcrumb/demo/component-token.tsx:43]: 最好能把所有 token 会改变的场景都列出来，现在 demo 里只能提现其中几个 token 的影响

Comment (heiyu4585) [reply components/breadcrumb/demo/component-token.tsx:80]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**